### ISSUE TITLE
Consider publishing headers in wheels to support external mixed C++/python module development.

### ISSUE THREAD

**Issue by Clockwork-Muse** [panda3d/panda3d#1223]
## Description
I'm fuzzy on whether this would be possible or would work the way I envision, but currently in order to create a module written in C++ that is exposed in python you must install _both_ the python package and the native sdk (ignoring other requirements for cross builds).  It would be handy if there was a wheel that included the required headers, so that only python packages would be required.

## Use Case
This may considerably simplify module development, by not requiring a separate sdk build, including possibly requiring a local build in some scenarios.

Python packages can require dependencies at build-time only, meaning a separate package is probably the best way (so, `panda3d-dev`?), although it's possible that the includes could just be excluded at application build time.

**Comment by rdb** [comment]
We'd not just need to provide headers, but also provide unstripped libraries, library symlinks (on Linux) and .lib files (on Windows). I suspect it would significantly increase the size of the wheels.

Perhaps a good alternative would be to provide them in a separate wheel?  Or to provide a script that downloads them?

**Comment by Clockwork-Muse** [comment]
Right, I specifically call out a separate wheel as an option (also - would that mean we'd want a separate dev wheel for the fully optimized one as well?  Obviously we don't want to distribute the dev portion in a compiled app, but we'd still be required to compile against the optimized one during module build).

A script as a tool we include with the wheel is probably also viable (ie, as another tool like `pview`, etc).  We'd need to interface with C++ build tools in either case, and that may make it easier for pulling the dependencies.  

On the other hand, if we can get module development to essentially just work off of a standard python module template, it may work better as a normal python (build time) dependency - this is part of why I want to look at things like more common C++->Python binding tools, because most of them have known routes to compiling C++ code into wheels as an automatic step.
